src/sample_weight_audit/estimator_check.py

```diff
@@ -255,50 +282,29 @@ def multifit_over_weighted_and_repeated(
         extra_params_repeated = {}
 
     # Perform one reference fit to inspect the predictions dimensions.
-    est_ref = clone(est).set_params(random_state=0, **extra_params_weighted)
+    est_ref = clone(est).set_params(**extra_params_weighted)
+    if "random_state" in signature(est.fit).parameters:
+        est_ref.set_params(random_state=0)
     est_ref = check_pipeline_and_fit(est_ref, X_train, y_train, sample_weight_train)
 
-    predictions_ref = compute_predictions(est_ref, X_test[:1])
-
-    # Adjust the number of predictions so that stat_test_dim = n_test_data_points *
-    # prediction_dim for all evaluated models. This is necessary to be able to
-    # compare the p-values across different models.
-    assert predictions_ref.ndim == 2
-    assert predictions_ref.shape[0] == 1
-    prediction_dim = predictions_ref.shape[1]
-
-    # Reduce the dimensionality of the predictions if necessary.
-    if prediction_dim * 3 > stat_test_dim:
-        # Reduce projection_dim such that test_size >= 3 to be able to
-        # minimally characterize the variations of the learned prediction
-        # function.
-        assert stat_test_dim % 3 == 0
-        prediction_dim = stat_test_dim // 3
-        rp = GaussianRandomProjection(
-            n_components=prediction_dim, random_state=random_state
-        )
-        rp.fit(predictions_ref)
-        project = rp.transform
-    else:
-        project = lambda x: x  # noqa: E731
-
-    n_test_data_points = stat_test_dim // prediction_dim
-
-    # If the following does not hold, we should raise an informative error message to tell
-    assert n_test_data_points <= X_test.shape[0]
-
-    X_test_diverse_subset = get_diverse_subset(
-        X_test, y_test, sample_weight_test, test_size=n_test_data_points
-    )
+    scores_weighted_all = []
+    scores_repeated_all = []
     predictions_weighted_all = []
     predictions_repeated_all = []
     for seed in tqdm(range(n_stochastic_fits)):
-        est_weighted = clone(est).set_params(random_state=seed, **extra_params_weighted)
-        # Use different random seeds for the other group of fits to avoid
-        # introducing an unwanted statistical dependency.
-        est_repeated = clone(est).set_params(
-            random_state=seed + n_stochastic_fits, **extra_params_repeated
-        )
+        if "random_state" in signature(est.__init__).parameters:
+            est_weighted = clone(est).set_params(
+                random_state=seed, **extra_params_weighted
+            )
+            # Use different random seeds for the other group of fits to avoid
+            # introducing an unwanted statistical dependency.
+            est_repeated = clone(est).set_params(
+                random_state=seed + n_stochastic_fits, **extra_params_repeated
+            )
+        else:
+            est_weighted = clone(est).set_params(**extra_params_weighted)
+            est_repeated = clone(est).set_params(**extra_params_repeated)
```

Comment:
Thinking about it, it's a bit of a waste to fit an estimator with no `random_state` parameter 100x times. We should probably optimize this by fitting them only once and using `pvalue = float(np.allclose(score_weighted, score_repeated))` and hard-code the deterministic flag.